src/bridge/cockpitpeer.c

```diff
@@ -262,9 +289,41 @@ on_other_control (CockpitTransport *transport,
           g_message ("%s: received \"authorize\" request without a valid cookie", self->name);
         }
 
-      /* If we have info we can respond to basic authorize challenges */
-      else if (cockpit_json_get_string (options, "challenge", NULL, &challenge) &&
-               challenge && g_hash_table_contains (self->authorize_values, challenge))
+      else if (!cockpit_json_get_string (options, "challenge", NULL, &challenge))
+        {
+          g_message ("%s: received \"authorize\" request with a invalid challenge", self->name);
+        }
+      else if (!cockpit_json_get_string (options, "prompt", NULL, &prompt))
+        {
+          g_message ("%s: received \"authorize\" request with a invalid prompt", self->name);
+        }
+
+      /* Hook into the superuser startup mechanism for privileged bridges.
+       */
+      else if (cockpit_json_get_bool (self->config, "privileged", FALSE, &privileged) && privileged)
+        {
+          if (self->startup_auth_cookie)
+            g_message ("%s: received overlapping \"authorize\" requests", self->name);
+          else if (!self->router)
+            g_message ("%s: no router for answering \"authorize\" request", self->name);
```

Comment:
Can these happen in practice? If not, these should be `g_warning()`, so that our tests will spot them.

Reply:
Good point. The former can happen when stuff happens concurrently that shouldn't. The latter can only happen with a custom bridge that doesn't do the initialization correctly. So, both fixed to be warnings.

Reply:
But using that logic, things like
```
g_message ("%s: received \"authorize\" request without a valid cookie", self->name);
```
should also be a warning, no?